### DISCUSSION TITLE
Category context is not provided.

### DISCUSSION THREAD

**Discussion by maximilian-schwarz** [shopware/frontends#58] [Q&A] [answered]
Hello,

currently i try the shopware/frontends with Nuxt 3. First of all it's really amazing. Thanks for the hard work.

I try the following example from the docs:
```js
<template>
  <div>
    <div v-for="product in getElements" :key="product.id">
      {{ product.name }}
    </div>
  </div>
</template>

<script setup lang="ts">
const { refreshSessionContext } = useSessionContext();

onMounted(() => {
  refreshSessionContext();
});

const { search, getElements } = useListing({
  listingType: "categoryListing",
  categoryId: "dfd52ab937f840fd87e9d24ebf6bd245"
});
</script>
```

**I got always the following error:**
Category context is not provided.

How i got the category context? Thanks for your help :)

**Comment by niklaswolf** [comment, marked as answer]
That's odd because looking at the composable code, if there is no context provided it should use the categoryId provided in the params...
Do you use the latest version of the composables package?

**Reply by patzick** [reply]
@maximilian-schwarz - as @niklaswolf mentioned, this should not have happened in your case. Could you prepare a reproduction of that using the Stackblitz template?
https://stackblitz.com/github/shopware/frontends/tree/main/templates/vue-demo-store

**Reply by maximilian-schwarz** [reply]
How can i create the correct category context via ID?

@patzick: I have create a Stackblitz example for you:
https://stackblitz.com/edit/nuxt-starter-fungdb?file=app.vue

First of all i refresh the session context. And with this part of code i want all product of a specific category:

```
const { getElements: products } = useListing({
  listingType: 'categoryListing',
  categoryId: 'ed1fa67623064e6e970d0e8725df28fd',
});
```

**Reply by maximilian-schwarz** [reply]
> That's odd because looking at the composable code, if there is no context provided it should use the categoryId provided in the params...
> Do you use the latest version of the composables package?

Yes i use the latest version of the composables package.

**Reply by patzick** [reply]
@maximilian-schwarz sorry it took that long to look into that. The problem was as initially written - old dependencies.
I upgraded the example to the newest canary (also to include settings changes), and the example is not throwing that error anymore.

Please compare your version to: https://stackblitz.com/edit/nuxt-starter-eybh8m?file=app.vue,package.json and let me know if that helps you.

**Reply by maximilian-schwarz** [reply]
@patzick Thanks for your support. With the canary version everything is working :)
